libs/contrib/Data/String/Parser.idr

```diff
@@ -121,6 +122,17 @@ export
 optional : Functor m => ParseT m a -> ParseT m (Maybe a)
 optional = optionMap Nothing Just
 
+||| Succeeds if and only if the argument parser fails.
+|||
+||| In Parsec, this combinator is called `notFollowedBy`.
+export
+requireFailure : Functor m => ParseT m a -> ParseT m ()
+requireFailure (P runParser) = P $ \s => reverseResult s <$> runParser s
+where
+  reverseResult : State -> Result a -> Result ()
+  reverseResult  s             (Fail _ _) = OK () s
+  reverseResult (S _ _ maxPos) (OK _ _)   = Fail maxPos "Purposefully changed OK to Fail" -- not sure about maxPos
```

Comment:
I think you should use `pos` here: the failure is at the given position, not
the end of the file. I would also write this as

```suggestion
  reverseResult s (OK _ _)   = Fail (pos s) "Purposefully changed OK to Fail"maxPos
```

so that it's robust to us changing the definition of `State`.

Reply:
Or even `s.pos`

Reply:
Sure. Given that some of the postfix syntax changes have been reverted, I am
not sure what parses and what does not so I was overly cautious in my suggestion.

Reply:
I went with `(pos s)` just to avoid any postfix issues, plus it's what I'm used to let. Let me know if I should change it.